src/refmvs.rs

```diff
@@ -879,17 +882,19 @@ pub(crate) unsafe fn rav1d_refmvs_find(
             &mut have_newmv,
             &mut have_row_mvs,
         ) as c_uint;
-        b_top
     } else {
-        ptr::null()
-    };
+        max_rows = 0;
+        n_rows = !0;
+        b_top = ptr::null();
+    }
 
     // left
-    let mut max_cols = 0;
-    let mut n_cols = !0;
-    let b_left = if bx4 > rt.tile_col.start {
+    let max_cols;
+    let n_cols;
+    let b_left;
+    if bx4 > rt.tile_col.start {
         max_cols = cmp::min(bx4 - rt.tile_col.start + 1 >> 1, 2 + (bw4 > 1) as c_int) as c_uint;
-        let b_left = &rt.r[(by4 as usize & 31) + 5];
+        b_left = rt.r[(by4 as usize & 31) + 5..].as_ptr();
```

Comment:
This change seems incorrect to me, though maybe I'm misunderstanding something. Previously `b_left` was set to one of the pointers in `rt.r`, but now it's going to be a pointer to `rt.r` itself. It doesn't look like you've changed how `b_left` is being used in this commit, so it seems like the usages of `b_left` are now going to be getting an incorrect pointer.

Reply:
Previously the `&` decayed the reference into a pointer, which was then used as a slice.  So I think that should be preserved, just done more correctly here with a slice and `.as_ptr()` in this commit, and then later into a `&[usize]` of indices.

Reply:
Ah I see, thanks for clarifying.